kornia/augmentation/utils/helpers.py

```diff
@@ -226,51 +226,60 @@ def _adapted_uniform(
     shape: Union[Tuple, torch.Size],
     low: Union[float, int, torch.Tensor],
     high: Union[float, int, torch.Tensor],
-    same_on_batch=False
+    same_on_batch: bool = False
 ) -> torch.Tensor:
     r"""The uniform sampling function that accepts 'same_on_batch'.
 
     If same_on_batch is True, all values generated will be exactly same given a batch_size (shape[0]).
     By default, same_on_batch is set to False.
+
+    By default, sampling happens on the default device and dtype. If low/high is a tensor, sampling will happen
+    in the same device/dtype as low/high tensor.
     """
     device, dtype = _extract_device_dtype([
         low if isinstance(low, torch.Tensor) else None,
         high if isinstance(high, torch.Tensor) else None,
     ])
     if not isinstance(low, torch.Tensor):
-        low = torch.tensor(low, device=device, dtype=torch.float64)
+        low = torch.tensor(float(low), device=device, dtype=dtype)
```

Comment:
I think in similar places like this one we could start using `torch.as_tensor`
https://pytorch.org/docs/stable/generated/torch.as_tensor.html?highlight=as_tensor#torch.as_tensor

we need to verify backward compatibility though

Reply:
Great. It should be okay for the earlier versions (it was there after 1.2.0).